Web.Tests/AggregationTests.cs

```diff
@@ -0,0 +1,490 @@
+            services.AddHttpClient();
+            serviceProvider = services.BuildServiceProvider();
+
+            clientFactory = serviceProvider.GetRequiredService<IHttpClientFactory>();
+        }
+        
+        [Theory]
+        [InlineData("16324")]
+        [InlineData("16325")]
+        public async void AverageOnSalesAmount(string port)
+        {
+            Test(await Get<dynamic>("/Sales?$apply=aggregate(Amount with average as AverageAmount)", port));
+
+            static void Test(ICollection<dynamic> collection)
+            {
+                Assert.Equal(1, collection.Count);
+                Assert.Equal(3, collection.First().AverageAmount.ToObject<decimal>());
+            }
+        }
+
+        [Theory]
+        [InlineData("16324")]
+        [InlineData("16325")]
+        public async void SumOnSalesAmount(string port)
+        {
+            Test(await Get<dynamic>("/Sales?$apply=aggregate(Amount with sum as SumAmount)", port));
+
+            static void Test(ICollection<dynamic> collection)
+            {
+                Assert.Equal(1, collection.Count);
+                Assert.Equal(24, collection.First().SumAmount.ToObject<decimal>());
+            }
+        }
+
+        [Theory]
+        [InlineData("16324")]
+        [InlineData("16325")]
+        public async void MinOnSalesAmount(string port)
+        {
+            Test(await Get<dynamic>("/Sales?$apply=aggregate(Amount with min as MinAmount)", port));
+
+            static void Test(ICollection<dynamic> collection)
+            {
+                Assert.Equal(1, collection.Count);
+                Assert.Equal(1, collection.First().MinAmount.ToObject<decimal>());
+            }
+        }
+
+        [Theory]
+        [InlineData("16324")]
+        [InlineData("16325")]
+        public async void MaxOnSalesAmount(string port)
+        {
+            Test(await Get<dynamic>("/Sales?$apply=aggregate(Amount with max as MaxAmount)", port));
+
+            static void Test(ICollection<dynamic> collection)
+            {
+                Assert.Equal(1, collection.Count);
+                Assert.Equal(8, collection.First().MaxAmount.ToObject<decimal>());
+            }
+        }
+
+        [Theory(Skip = "Navigation properties cannot be part of the '$apply'-option (not yet supported)")]
+        [InlineData("16324")]
+        [InlineData("16325")]
+        public async void AverageOnProductTaxRate(string port)
+        {
+            Test(await Get<dynamic>("/Sales?$apply=aggregate(Product/TaxRate with average as AverageProduct_TaxRate)", port));
+
+            static void Test(ICollection<dynamic> collection)
+            {
+                Assert.Equal(1, collection.Count);
+                Assert.Equal(0.1, collection.First().AverageProduct_TaxRate.ToObject<decimal>());
+            }
+        }
+
+        [Theory(Skip = "Navigation properties cannot be part of the '$apply'-option (not yet supported)")]
```

Comment:
Questions:  What's the reason for "Navigation properties not supported" and what would a future fix be?

Reply:
The reason is quite simple; in the `$apply` a `linq-to-sql` is performed which doesn't have knowledge of the mapping of the models to the actual database entities. This will be resolved once the `$apply` is implemented in these extensions.

Reply:
Because we're all busy a PR should not lead to more open issues and discussions than before.  If an implementation is still needed then maybe we should just do that if you're interested.  See some of the existing code in[ FilterHelper](https://github.com/AutoMapper/AutoMapper.Extensions.OData/blob/52ff5963221b33a126365d402a3c5e8e5efb2b88/AutoMapper.AspNetCore.OData.EFCore/FilterHelper.cs#L44).  The [OData repository](https://github.com/OData/AspNetCoreOData/blob/230a84c36a04bdc835c173d2c128a3e860fc6817/src/Microsoft.AspNetCore.OData/Query/Expressions/AggregationBinder.cs#L28) is also a good resource for the business logic.

Reply:
This PR doesn't not lead to more open issues and discussions. The issues and discussions were already there. This PR explains how one can make use of the `EnableQueryAttribute` and how the `$apply`-option can work, but current there are some limits. The tests give some insights on those limits that need extra work.
In regard to the implementation question; yes, I'm interested, but I would like to finish this PR first 😉

Reply:
As it is I think the library is relatively simple to use.  May be better to PR a [separate branch](https://github.com/AutoMapper/AutoMapper.Extensions.OData/tree/EnableQuery) until it's ready?  We get questions even when we think the code is complete.

In this case it's unclear to me how you bridge the gap between the attribute and this library.  How to debug it etc..

Currently it's mostly separate - you can remove [EnableQueryFeatures here](https://github.com/AutoMapper/AutoMapper.Extensions.OData/blob/b44c99cfa0afdb2a1ab5a7f4d310ea0ae3f42d90/WebAPI.OData.EFCore/Startup.cs#L33) and it all still works.

Makes sense?

Reply:
I didn't find the time yet to look into your question, but I'm okay with pushing this PR to a separate branch. Based on what you stated I think that this would be the step that would make the most sense.

Reply:
Ok - created the [EnableQuery](https://github.com/AutoMapper/AutoMapper.Extensions.OData/tree/EnableQuery) branch.

Reply:
Thanks! I changed the destination branch for this PR to the branch [EnableQuery](https://github.com/AutoMapper/AutoMapper.Extensions.OData/tree/EnableQuery). If you want it can be merged into that branch